client/resteasy-client/src/main/java/com/cerner/beadledom/client/resteasy/BeadledomResteasyClientBuilder.java

```diff
@@ -300,7 +301,7 @@ public BeadledomResteasyClient build() {
     CorrelationIdContext correlationIdContext =
         new ResteasyCorrelationIdContext(
             clientConfig.correlationIdName(),
-            clientConfig.correlationIdName());
+            clientConfig.correlationIdName(), new ThreadLocalCorrelationContext());
```

Comment:
This will work since the `ThreadLocal` in the implementation is static, but do we have any thoughts on how we can keep this working if we were to use an implementation using opentelemetry? (e.g. how do we make sure the container implementation and the client implementation within that container match?)

Reply:
I probably should expose setting the correlationcontext on the builder object. But to your question, the consumer would need to bind to the same implementation. We could also use a CorrelationContextManager that could lookup the implementation based on SPI.